### PR TITLE
fix(cr)_: fix dismissing and then sending a CR not showing the message or updating the AC notif

### LINKED ISSUE #16817: Rejecting a Contact Request blocks further attempts to add user as a contact
# Bug Report

## Description

user 1 sends a contact request to user 2. User 2 rejects. The contact request gets stuck and these users cannot add each other.
There seems to be an issue triggered when the user rejects the contact request, blocking any other ways to add the contact.

## Steps to reproduce

1. Create a new user (user 1)
2. send a contact request to user 2
3. As user 2, reject the contact request

#### Expected behavior
As user 1, when going to Settings > Messaging > contacts > Pending Requests, the request sent to user 2 should be deleted

#### Actual behavior
As user 1, when going to Settings > Messaging > contacts > Pending Requests, the request sent to user 2 is still showing up. I've tested with 3 new users, and the behavior can be reproduced.

Also, if the user 2 sends a contact request to user 1:
- no notification is shown on user 1's end
- On user 1's end, the previous request sent by user 1 (step 2) is still there as a Pending Request.

If user 1 blocks and unblocks user 2 to remove them from the 'Pending Requests', when trying to add them as a contact, user 2 cannot add user 1.

### Additional Information

[test 054 logs.zip](https://github.com/user-attachments/files/17920725/test.054.logs.zip)
[test054 geth.log.zip](https://github.com/user-attachments/files/17920731/test054.geth.log.zip)

[test053 geth.log.zip](https://github.com/user-attachments/files/17920735/test053.geth.log.zip)
[test 053 logs.zip](https://github.com/user-attachments/files/17920744/test.053.logs.zip)

Image

- Status desktop version: latest master
- Operating System: Mac OS Sonoma. Also tested on Windows.

### PR DESCRIPTION
**Pull request by jrainville**
Fixes https://github.com/status-im/status-desktop/issues/16817

There were two issues.
When dismissing a CR, then sending one back, it did mark the two contacts as mutual and showed the 1-1 chat. However, the message sent in the second/final CR was not shown in the first person's client. Also, the AC notification for the first user didn't update, so it got stuck in a "pending" state.

Those two issues are fixed now with a test to confirm.

User 1 (og sender where issue appeared)
image

User 2 (the one that dismissed and then sent a new one)
image

### CODE REVIEW

**Comment by jrainville** [comment]
I ran the updated test 1000 times and it passed
```
PASS
ok      github.com/status-im/status-go/protocol 1466.124s
```

**Comment by jrainville** [comment]
it seems like this also fixed a problem in mobile :tada: https://github.com/status-im/status-mobile/issues/21438#issuecomment-2511403552

Reminder to review @igor-sirotin @osmaczko @qfrank

**Review by qfrank — APPROVED** [review]
Thank you for resolving this troublesome issue!

File: protocol/activity_center_persistence.go [hunk 1]
```diff
@@ -697,6 +697,52 @@ func (db sqlitePersistence) GetActivityCenterNotificationsByID(ids []types.HexBy
...
+			c.last_message,
+			a.reply_message,
+			a.contact_verification_status,
+			c.name,
+			a.author,
+			a.token_data,
+			a.updated_at,
+			a.installation_id
+		FROM activity_center_notifications a
+		LEFT JOIN chats c ON c.id = a.chat_id
```
Comment (igor-sirotin) [inline protocol/activity_center_persistence.go]: ouch, I see that not you started this, but can you please extract this mess to a constant?
Perhaps `protocol/persistence_metrics.go` can be an okay-ish example.

Or go with [Squirrel](https://github.com/Masterminds/squirrel), as here:
https://github.com/status-im/status-go/blob/cfcef92e64ad1c2e8c85cce6a668f9091c7d717f/transactions/pendingtxtracker_db.go#L43-L45

But I leave it at your decision if you want to refactor this now ofc.

File: protocol/messenger_handler.go [hunk 2]
```diff
@@ -2374,6 +2374,7 @@ func (m *Messenger) handleChatMessage(state *ReceivedMessageState, forceSeen boo
 		m.logger.Warn("failed to add peersyncing message", zap.Error(err))
 	}
 
+	receivedAContactRequest := false
```
Comment (igor-sirotin) [inline protocol/messenger_handler.go:2377]: hmm, what `A` means here? 🤔

**Review by igor-sirotin — APPROVED** [review]

File: protocol/messenger_handler.go (hunk 2 above)
Comment (jrainville) [reply protocol/messenger_handler.go:2377]: https://www.dictionary.com/browse/a

It's just `a contact request`. I guess it's not that necessary, but it's nothing special :joy:

**Review by osmaczko — APPROVED** [review]

**Pull request merged**